client/android-java/app/src/main/java/com/example/app/CheckoutActivity.java

```diff
@@ -0,0 +1,148 @@
+import okhttp3.Response;
+
+public class CheckoutActivity extends AppCompatActivity {
+    /**
+     * This example collects card payments, implementing the guide here: https://stripe.com/docs/payments/accept-a-payment#android
+     * <p>
+     * To run this app, follow the steps here: https://github.com/stripe-samples/mobile-elements-card-payment#how-to-run
+     */
+    // 10.0.2.2 is the Android emulator's alias to localhost
+    private static final String BACKEND_URL = "http://10.0.2.2:4242/";
+
+    private OkHttpClient httpClient = new OkHttpClient();
+    private Stripe stripe;
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_checkout);
+
+        // Configure the SDK with your Stripe publishable key so that it can make requests to the Stripe API
+        // ⚠️ Don't forget to switch this to your live-mode publishable key before publishing your app
+        PaymentConfiguration.init(getApplicationContext(), "Insert your publishable key"); // Get your key here: https://stripe.com/docs/keys#obtain-api-keys
+
+        // Hook up the pay button to the card widget and stripe instance
+        Button payButton = findViewById(R.id.payButton);
+        WeakReference<CheckoutActivity> weakActivity = new WeakReference<>(this);
+        payButton.setOnClickListener((View view) -> {
+            // Get the card details from the card widget
+            CardInputWidget cardInputWidget = findViewById(R.id.cardInputWidget);
+            Card card = cardInputWidget.getCard();
+            if (card != null) {
+                // Create a Stripe Token from the card details
+                stripe = new Stripe(getApplicationContext(), PaymentConfiguration.getInstance(getApplicationContext()).getPublishableKey());
+                stripe.createToken(card, new ApiResultCallback<Token>() {
+                    @Override
+                    public void onSuccess(@NonNull Token result) {
+                        // Send the token identifier to the server
+                        MediaType mediaType = MediaType.get("application/json; charset=utf-8");
+                        String json = "{"
+                                + "\"currency\":\"usd\","
+                                + "\"items\":["
+                                + "{\"id\":\"photo_subscription\"}"
+                                + "],"
+                                + "\"token\":\"" + result.getId() + "\""
+                                + "}";
+                        RequestBody body = RequestBody.create(json, mediaType);
+                        Request request = new Request.Builder()
+                                .url(BACKEND_URL + "create-payment-intent")
+                                .post(body)
+                                .build();
+                        httpClient.newCall(request)
+                                .enqueue(new Callback() {
```

Comment:
This line throws an exception: 
`    java.lang.RuntimeException: Can't create handler inside thread Thread[OkHttp http://10.0.2.2:4242/...,5,main] that has not called Looper.prepare()`

@mshafrir-stripe any idea why?  It looks identical to the kotlin code to me.

Reply:
`displayAlert()` is missing `runOnUiThread` - see https://github.com/stripe-samples/card-payment-charges-api/pull/2/files/54d0d5d43fc1ea82ddf54dc0280e4f9c4efb50bc#diff-e65748b842d1fc142e4a8d557cc64cf7R109